main/php_ini.c

```diff
@@ -410,6 +410,67 @@ static void php_load_zend_extension_cb(void *arg) { }
 #endif
 /* }}} */
 
+/* {{{ get_env_location
+ */
+static char* get_env_location(const char *envname)
+{
+	char *env_location = getenv(envname);
+#ifdef PHP_WIN32
+	char *reg_location;
+	char phprc_path[MAXPATHLEN];
+#endif
+
+#ifdef PHP_WIN32
+	if (!env_location) {
+		char dummybuf;
+		int size;
+
+		SetLastError(0);
+
+		/*If the given buffer is not large enough to hold the data, the return value is
+		the buffer size,  in characters, required to hold the string and its terminating
+		null character. We use this return value to alloc the final buffer. */
+		size = GetEnvironmentVariableA(envname, &dummybuf, 0);
+		if (GetLastError() == ERROR_ENVVAR_NOT_FOUND) {
+			/* The environment variable doesn't exist. */
+			return "";
+		}
+
+		if (size == 0) {
+			return "";
+		}
+
+		size = GetEnvironmentVariableA(envname, phprc_path, size);
```

Comment:
I think the `size` parameter should be `min(size, MAXPATHLEN)`, not to overflow `phprc_path`.

Reply:
@nikic fix this or leave as is?